ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -362,6 +399,192 @@ private void verifyCrc() throws MessageFormatException {
   }
 
   /**
+   *
+   *  - - - - - - - - - - - - - - - - - - -- - -- - - - - - - - - - - - - -  - - - - - - - - - - - - - - - - - - - -
+   * |         |              |                 |               |           |               |           |           |
+   * | version | payload size | Blob Encryption | Blob Property | Delete    | User Metadata | Blob      | Crc       |
+   * |(2 bytes)|   (8 bytes)  | Key Relative    | Relative      | Relative  | Relative      | Relative  | (8 bytes) |
```

Comment:
Can you remind me something. If user metadata is null for a blob on upload, how does disk format look like? Do we have empty userMetadataRecord or userMetadataRecord is optional in the disk format?

Reply:
Basically, I am trying to see if we should follow the same convention for KeyRecord as well.

Reply:
MessageFormatRecord does not accept null usermetadata, it accepts empty user metadata though. I would like to keep the encryption key as a truly optional thing.